go-app.js

```diff
@@ -232,14 +234,96 @@ go.app = function() {
     });
 
     self.states.add('Submit_Case', function(name) {
-      return new EndState(name, {
-        text: "Thank you! Your report has been submitted.",
-        next: 'Facility_Code_Entry'
-      });
+      return Q()
+        .then(function() {
+          // Send response to Ona
+          var ona_conf = self.im.config.ona;
+          if (typeof ona_conf == 'undefined') {
+            return self.im.log.info([
+              "No Ona API configured.",
+              "Not submitting data to Ona."
+            ].join(" "));
+          }
+          var ona = new Ona(self.im, {
+            auth: {
+              username: ona_conf.username,
+              password: ona_conf.password
+            },
+            url: ona_conf.url
+          });
+          var data = self.im.user.answers;
+          data.create_date_time = new Date();
+          data.reported_by = self.im.user.addr;
+          data.gender = (self.im.user.answers.No_SA_ID_Gender_Entry == 1) ? "male" : "female";
+          said = data.SA_ID_Entry;
+          if (data.SA_ID_Entry) {
+            data.No_SA_ID_Year_Entry = said.substring(0, 2);
+            data.No_SA_ID_Month_Entry = said.substring(2, 4);
+            data.No_SA_ID_Day_Entry = said.substring(4, 6);
+          }
```

Comment:
Can you explain what's going on here? What's the relation between `SA_ID_Entry` and year, month & day? Is that information encoded in the SA ID?